model/integration_action_test.go

```diff
@@ -155,3 +156,120 @@ func TestPostActionIntegrationEquals(t *testing.T) {
+		for i := 0; i < nb; i++ {
+			elements = append(elements, DialogElement{DisplayName: "display", Name: fmt.Sprintf("Name %d", i), Type: "text", SubType: "password", Default: "default", Placeholder: "placeholder", HelpText: "help text", Optional: false, MinLength: 1, MaxLength: 150, DataSource: "", Options: nil})
+		}
+		return elements
+	}
+	invalidDialogElement := DialogElement{}
+
+	testCases := []validableTestCase{
+		{name: "callback id empty", validable: Dialog{CallbackId: "", Title: "title", IntroductionText: "intro text", IconURL: "http://mattermost.com/icon.svg", Elements: makeValidDialogElementArray(1)}, hasError: true},
+		{name: "title empty", validable: Dialog{CallbackId: "callbackid", Title: "", IntroductionText: "intro text", IconURL: "http://mattermost.com/icon.svg", Elements: makeValidDialogElementArray(1)}, hasError: true},
+		{name: "title too long", validable: Dialog{CallbackId: "callbackid", Title: "ABCDEFGHIJKLMNOPQRSTUVWXYZ", IntroductionText: "intro text", IconURL: "http://mattermost.com/icon.svg", Elements: makeValidDialogElementArray(1)}, hasError: true},
+		{name: "icon url invalid", validable: Dialog{CallbackId: "callbackid", Title: "title", IntroductionText: "intro text", IconURL: "mattermost.com/icon.svg", Elements: makeValidDialogElementArray(1)}, hasError: true},
+		{name: "two elements with same name", validable: Dialog{CallbackId: "callbackid", Title: "title", IntroductionText: "intro text", IconURL: "http://mattermost.com/icon.svg", Elements: []DialogElement{{Name: "same name", DisplayName: "Display", Type: "text", MaxLength: 5}, {Name: "same name", DisplayName: "Display", Type: "text", MaxLength: 5}}}, hasError: true},
+		{name: "too many elements", validable: Dialog{CallbackId: "callbackid", Title: "title", IntroductionText: "intro text", IconURL: "http://mattermost.com/icon.svg", Elements: makeValidDialogElementArray(6)}, hasError: true},
+		{name: "invalid element", validable: Dialog{CallbackId: "callbackid", Title: "title", IntroductionText: "intro text", IconURL: "http://mattermost.com/icon.svg", Elements: []DialogElement{invalidDialogElement}}, hasError: true},
+		{name: "all valid", validable: Dialog{CallbackId: "callbackid", Title: "title", IntroductionText: "intro text", IconURL: "http://mattermost.com/icon.svg", Elements: makeValidDialogElementArray(1)}, hasError: false},
+	}
+
+	validableTestcasesRun(t, testCases)
+}
+
+func TestDialogElementIsValid(t *testing.T) {
+	genLongString := func(length int) string {
+		s := ""
+		for i := 0; i < length; i++ {
+			s += "a"
+		}
+		return s
+	}
+
+	testCases := []validableTestCase{
+		// general
+		{name: "general: DisplayName empty", validable: DialogElement{DisplayName: "", Name: "name", Type: "text", SubType: "password", Default: "default", Placeholder: "placeholder", HelpText: "help text", Optional: false, MinLength: 0, MaxLength: 3000, DataSource: "", Options: nil}, hasError: true},
```

Comment:
Same suggestion here to use a helper function to clean up the test data duplication here